google/cloud/pubsublite/integration_tests/publisher_integration_test.cc

```diff
@@ -0,0 +1,171 @@
+                "pubsublite.googleapis.com"))} {
+    auto topic_id = RandomTopicName();
+
+    GarbageCollect();
+
+    CreateTopicRequest req;
+    req.set_parent("projects/" + project_id_ + "/locations/" + location_id_);
+    req.set_topic_id(topic_id);
+    req.mutable_topic()->mutable_partition_config()->set_count(3);
+    req.mutable_topic()
+        ->mutable_partition_config()
+        ->mutable_capacity()
+        ->set_publish_mib_per_sec(kThroughputCapacityMiB);
+    req.mutable_topic()
+        ->mutable_partition_config()
+        ->mutable_capacity()
+        ->set_subscribe_mib_per_sec(kThroughputCapacityMiB);
+    req.mutable_topic()->mutable_retention_config()->set_per_partition_bytes(
+        kPartitionStorage);
+    EXPECT_STATUS_OK(admin_connection_->CreateTopic(std::move(req)));
+    auto topic = Topic{project_id_, location_id_, topic_id};
+    topic_name_ = topic.FullName();
+    publisher_ = *MakePublisherConnection(topic, Options{});
+  }
+
+  ~PublisherIntegrationTest() override {
+    DeleteTopicRequest req;
+    req.set_name(topic_name_);
+    admin_connection_->DeleteTopic(std::move(req));
+  }
+
+  void GarbageCollect() {
+    ListTopicsRequest req;
+    req.set_parent("projects/" + project_id_ + "/locations/" + location_id_);
+    auto topics = admin_connection_->ListTopics(std::move(req));
+    std::string full_topic_prefix =
+        "projects/" + project_id_ + "/locations/" + location_id_ + "/topics/" +
+        "pub-int-test-" +
+        google::cloud::internal::FormatUtcDate(
+            std::chrono::system_clock::now() - std::chrono::hours(48)) +
+        "-";
+    for (auto const& topic : topics) {
+      if (!std::regex_search(topic->name(), topic_regex_)) continue;
+      if (topic->name() < full_topic_prefix) {
+        DeleteTopicRequest delete_req;
+        delete_req.set_name(topic->name());
+        admin_connection_->DeleteTopic(std::move(delete_req));
+      }
+    }
+  }
+
+  std::string RandomTopicName() {
+    auto generator =
+        google::cloud::internal::DefaultPRNG(std::random_device{}());
+    return topic_prefix_ +
+           google::cloud::internal::Sample(
+               generator, 42, "abcdefghijlkmnopqrstuvwxyz0123456789_-") +
+           google::cloud::internal::Sample(
+               generator, 1, "abcdefghijlkmnopqrstuvwxyz0123456789");
+  }
+
+  std::chrono::system_clock::time_point tp_;
+  std::regex topic_regex_ = std::regex{
+      R"re(^projects\/\d*\/locations\/[a-z0-9\-]*\/topics\/pub-int-test[-_]\d{4}[-_]\d{2}[-_]\d{2}[-_])re"};
+  std::string topic_prefix_;
+  std::string project_id_;
+  std::string location_id_;
+  std::shared_ptr<AdminServiceConnection> admin_connection_;
+  std::string topic_name_;
+  std::unique_ptr<PublisherConnection> publisher_;
+};
+
+TEST_F(PublisherIntegrationTest, BasicGoodWithoutKey) {
+  std::vector<future<StatusOr<std::string>>> results;
+  for (unsigned int i = 0; i != kNumMessages; ++i) {
```

Comment:
I missed these earlier.  Do they need to be `unsigned int i` and `unsigned i`?  Why not just `int`?

Reply:
typo
`int` is fine